postgresql/resource_postgresql_default_privileges.go

```diff
@@ -110,6 +112,18 @@ func resourcePostgreSQLDefaultPrivilegesCreate(d *schema.ResourceData, meta inte
 	}
 	defer deferredRollback(txn)
 
+	// Needed in order to set the owner of the db if the connection user is not a
+	// superuser
+	ownerGranted, err := grantRoleMembership(client.DB(), owner, currentUser)
```

Comment:
In this case I think it would be great to use `txn` instead of creating a new connection (and so you can't use `defer` to call `revokeRoleMembership` but you can call it just before the `txn.Commit();`

Thanks to that this role will not be granted outside of the transaction.

You just need the type of the first parameter in both  `grantRoleMembership` and `revokeRoleMembership` to accept `sql.Tx` or `sql.Db`.

I already wrote in another (not yet published) PR an interface for that (inspired from an issue in the `database/sql` lib):
```
// QueryAble is common interface for sql.DB and sql.Tx
// (cf https://github.com/golang/go/issues/14468)
type QueryAble interface {
    Exec(query string, args ...interface{}) (sql.Result, error)
    Query(query string, args ...interface{}) (*sql.Rows, error)
    QueryRow(query string, args ...interface{}) *sql.Row
}

[...]

func grantRoleMembership(db QueryAble, [...])
```

Reply:
I made it in the same way we already do that GRANT, I guess refactoring this to use `txn` could be made in another PR ?